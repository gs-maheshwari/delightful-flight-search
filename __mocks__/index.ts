export const mockAirports = [
  {
    ItemName: "AMS",
    AirportName: "Amsterdam",
    Description: "Amsterdam, Netherlands",
  },
  {
    ItemName: "BUD",
    AirportName: "Budapest",
    Description: "Budapest, Hungary",
  },
];


export const mockFlightOffer = [
    {
      outboundFlight: {
        id: "AMSBUD20221110HV9911",
        departureDateTime: "2022-11-10T08:00:00",
        arrivalDateTime: "2022-11-10T09:55:00",
        marketingAirline: {
          companyShortName: "HV",
        },
        flightNumber: 9911,
        departureAirport: {
          locationCode: "AMS",
        },
        arrivalAirport: {
          locationCode: "BUD",
        },
      },
      pricingInfoSum: {
        totalPriceAllPassengers: 64.17,
        totalPriceOnePassenger: 64.17,
        baseFare: 27.13,
        taxSurcharge: 37.04,
        currencyCode: "EUR",
        productClass: "Basic",
      },
      deeplink: {
        href: "https://www.transavia.com/nl-NL/bookingtool/flights/deeplink?ds=AMS&as=BUD&ap=1&cp=0&od=10&om=11&oy=2022&utm_source=API&utm_medium=Public GOS",
      },
    },
  ];
