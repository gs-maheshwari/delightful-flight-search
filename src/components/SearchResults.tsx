import type { Airport, Flight } from "../types";

interface SearchResultsProps {
  flightOffer: Flight[];
  airports: Airport[];
}

const SearchResults = ({ flightOffer = [], airports = [] }: SearchResultsProps) => {
  const getAirportName = (code: string) => {
    const airport = airports.find((a) => a.ItemName === code);
    return airport ? airport.AirportName : code;
  };

  return (
    <div>
      <h2 className="mb-4 text-2xl font-bold text-green-600">Search Results</h2>
      {flightOffer.length === 0 ? (
        <p>No flights found for the selected criteria.</p>
      ) : (
        <ul className="space-y-4">
          {flightOffer.map((flight: Flight) => {
            const { outboundFlight, pricingInfoSum } = flight;
            const {
              id,
              flightNumber,
              departureAirport,
              arrivalAirport,
              departureDateTime,
              arrivalDateTime,
              marketingAirline,
            } = outboundFlight;
            return (
              <li key={id} data-testid={id} className="rounded-md border p-4 shadow-md">
                <div className="mb-2 flex items-center justify-between">
                  <div>
                    <span className="font-bold">
                      {getAirportName(departureAirport.locationCode)}
                    </span>
                    {" â†’ "}
                    <span className="font-bold">
                      {getAirportName(arrivalAirport.locationCode)}
                    </span>
                  </div>
                  <div className="font-bold text-green-600">
                    <span data-testid={`${id}-price`}>{pricingInfoSum.totalPriceAllPassengers.toFixed(2)}</span>
                    &nbsp;{pricingInfoSum.currencyCode}
                  </div>
                </div>
                <div className="text-sm text-gray-600">
                  <p>
                    Departure: {new Date(departureDateTime).toLocaleString()}
                  </p>
                  <p>Arrival: {new Date(arrivalDateTime).toLocaleString()}</p>
                  <p>
                    Flight: {marketingAirline.companyShortName} {flightNumber}
                  </p>
                </div>
              </li>
            );
          })}
        </ul>
      )}
    </div>
  );
};

export default SearchResults;
