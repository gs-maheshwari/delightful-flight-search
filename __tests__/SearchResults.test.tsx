import React from "react";
import { render, screen } from "@testing-library/react";
import SearchResults from "@/components/SearchResults";
import { mockAirports, mockFlightOffer } from '../__mocks__'

describe("SearchResults", () => {
  it("renders the search results title", () => {
    render(<SearchResults flightOffer={[]} airports={[]} />);
    expect(screen.getByRole("heading", { level: 2 })).toBeInTheDocument();
  });

  it("displays a message when no flights are found", () => {
    render(<SearchResults flightOffer={[]} airports={[]} />);
    expect(
      screen.getByText("No flights found for the selected criteria.")
    ).toBeInTheDocument();
  });

  it("renders flight information when flights are available", () => {
    render(
      <SearchResults flightOffer={mockFlightOffer} airports={mockAirports} />
    );

     const flightResultList = screen.getByTestId("AMSBUD20221110HV9911");
     expect(flightResultList).not.toBeEmptyDOMElement();

     const price = screen.getByTestId("AMSBUD20221110HV9911-price");
     expect(price).toHaveTextContent('64.17');
  });

  it('renders results unchanged', () => {
    const { container } = render(<SearchResults flightOffer={mockFlightOffer} airports={mockAirports}  />)
    expect(container).toMatchSnapshot()
  })
});
