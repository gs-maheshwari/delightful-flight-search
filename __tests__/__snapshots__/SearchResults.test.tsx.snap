// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders results unchanged 1`] = `
<div>
  <div>
    <h2
      class="mb-4 text-2xl font-bold text-green-600"
    >
      Search Results
    </h2>
    <ul
      class="space-y-4"
    >
      <li
        class="rounded-md border p-4 shadow-md"
        data-testid="AMSBUD20221110HV9911"
      >
        <div
          class="mb-2 flex items-center justify-between"
        >
          <div>
            <span
              class="font-bold"
            >
              Amsterdam
            </span>
             → 
            <span
              class="font-bold"
            >
              Budapest
            </span>
          </div>
          <div
            class="font-bold text-green-600"
          >
            <span
              data-testid="AMSBUD20221110HV9911-price"
            >
              64.17
            </span>
             
            EUR
          </div>
        </div>
        <div
          class="text-sm text-gray-600"
        >
          <p>
            Departure: 
            10/11/2022, 08:00:00
          </p>
          <p>
            Arrival: 
            10/11/2022, 09:55:00
          </p>
          <p>
            Flight: 
            HV
             
            9911
          </p>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
